### PR TITLE
Introduce `mike` to the doc generation process.

### PR DESCRIPTION
**Pull request by mhammond**
This causes the version selector to be rendered into the pages, however they will not be shown until we actually deploy version information.

Includes a script to help generate redirects for all our "old" pages.

On the path to #2016. Actual deployment using this system to follow once these are landed.

### CODE REVIEW

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:12]:
```suggestion
  the versioning system we have in place.**
```

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:24]: I think  those should be slightly restructured so it's easier to follow.

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:26]: If this is not inline code but a block it might be easier to copy (double click) vs needing to carefully select the text.

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:29]: I don't understand this. Is this done automatically or is it not?
If it is, it shouldn't be in this section of the instructions. Let's focus on the manual steps that _need_ to be done.

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:31]: Why is this a manual step btw? Why is this not done on CI?

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:50]:
```suggestion
* `./versions.json` is a file maintained by `mike` and used in the generated
  html as a version selector.

* `./next` is the directory where `mike` publishes the docs on currrent git `main`.

* `./0.27`, `./0.28` etc are directories where `mike` publishes released versions.

* `./latest` is a symlink managed by `mike` which points to the most recently released version.
```

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:14]:
```suggestion
* We use a [`mike`](https://github.com/jimporter/mike) to manage
```

File: docs/manual/src/README.md
Comment (badboy) [inline docs/manual/src/README.md:52]: I do not understand this sentence. Who edits the other content?

File: tools/make_old_docs_symlinks.py
Comment (badboy) [inline tools/make_old_docs_symlinks.py:4]: Does that mean this can go away once we did the full migration?

**Review by badboy — CHANGES_REQUESTED** [review]

File: docs/manual/src/README.md
Comment (mhammond) [reply docs/manual/src/README.md:31]: Because I don't want to block on that for landing this. I'd love it if you helped do this after landing.

File: docs/manual/src/README.md
Comment (mhammond) [reply docs/manual/src/README.md:29]: Yes, pushing docs for the main branch is on CI, pushing docs for a just-released version is not. But sure, I'll remove this.

File: docs/manual/src/README.md
Comment (mhammond) [reply docs/manual/src/README.md:52]: Us. Anything in the root of the directory or otherwise not published by mike in a versioned directory is maintained by the uniffi maintainers manually.

File: tools/make_old_docs_symlinks.py
Comment (mhammond) [reply tools/make_old_docs_symlinks.py:4]: As per the comment, it might make sense to keep it around so the top-level html redirects can be bumped to a later version. If we think we never ever want to move them from 0.27 it could be removed. In fact it could be removed now, because I've already run it, but am yet to push it to gh-pages.

File: docs/manual/src/README.md
Comment (mhammond) [reply docs/manual/src/README.md:52]: I tried to make that clearer.

File: docs/manual/src/README.md
Comment (badboy) [reply docs/manual/src/README.md]: Got it! Just wanted to make sure there's nothing fundamentally blocking this. Happy to help in a followup.

**Review by badboy — APPROVED** [review]

**Pull request merged**
